Add client graphql mutation to store data on server

nilsjung opened the pull request:
## for testing
you have to edit the current address in `frontend/client/index.js` to the ip of your local machine. 
also make sure, that the _mongodb_ and _server_ are running and listening on the right port and address.

Commit:
Added mocha/chai testing for mongo database

Commit:
Extend data structur

Commit:
Add query all function

Commit:
Merge branch 'master' into development

Commit:
Add mudation and change test framework

Commit:
Add makefile for backend service

Commit:
Client can save 'currentInformation' from store on server

Commit:
merge master

Commit:
Change button behaviour

karsil commented on frontend/reducers/index.js:
```diff
@@ -1,25 +1,50 @@
+
+    return {
+        ...state,
+        data: addElementToArray(elements, state.data),
```
Warning `[Unhandled promise rejection: TypeError: Invalid attempt to spread non-iterable instance]`

karsil commented on frontend/reducers/index.js:
```diff
@@ -1,25 +1,50 @@
+}
+
+function addElementToArray(element, array) {
+    let elements = [...array];
```
Warning, see above

Commit:
Refactor array.push function behaviour

Commit:
Add catch to mutation request

Commit:
Remove log

Commit:
fix database url and state manipulation

Commit:
Merge master

Commit:
Remove console log

nilsjung commented:
* I fixed the warnings and errors. This was a problem regarding the return value of the `Array.push()` function. This returns the new array length and not the new created array.
* I fixed the different graphql results that were stored local in the data-array. now they are consistent and can be stored safely with `store.data.push`
* I merge the actual master

Commit:
Fix autoloading

Commit:
Small fixes

* Map component stability
* state manipulation

Commit:
Fix reducer state manipulation

Commit:
Manipulate temp object in reducer

karsil reviewed (approved):
Approved with disclaimer to correspondig bug ticket (issue #22)
